common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisPilotingManager.java

```diff
@@ -834,8 +888,44 @@ public void setTargetLocation(TardisNavLocation targetLocation) {
         this.targetLocation = targetLocation;
     }
 
-    public void setCurrentLocation(TardisNavLocation currentLocation) {
-        this.currentLocation = currentLocation;
+    public void setCurrentLocationOnNextTick(ShellBaseBlockEntity blockEntity) {
+        this.newCurrentBlockEntity = blockEntity;
+    }
```

Comment:
I see what you're going for with this and I mostly just ask for it to be tidied up with variables
For example, calling getLevel on the same thing over and over should just be one variable, it means that performance is better by like 0.00000000001% but some players on terrible hardware need that

Reply:
I didn't realise that `getLevel()` did so much logic.
Is this better?